app/models/validation_request.rb

```diff
@@ -3,15 +3,20 @@
 module ValidationRequest
   extend ActiveSupport::Concern
 
+  class RecordCancelError < RuntimeError; end
+
+  class CancelledEmailError < StandardError; end
+
   included do
     include AASM
 
     aasm.attribute_name :state
 
-    aasm do
+    aasm whiny_persistence: true do
```

Comment:
https://www.rubydoc.info/github/aasm/aasm/AASM%2FConfiguration:whiny_persistence

Needed to raise error on bang! methods or it will just return false